command/agent/config/config.go

```diff
@@ -193,8 +188,19 @@ func LoadConfig(path string) (*Config, error) {
 		return nil, errwrap.Wrapf("error parsing 'vault':{{err}}", err)
 	}
 
-	if err := parseRetry(result, list); err != nil {
-		return nil, errwrap.Wrapf("error parsing 'retry': {{err}}", err)
+	if result.Vault == nil {
+		result.Vault = &Vault{}
+	}
+
+	// Set defaults
+	if result.Vault.Retry == nil {
+		result.Vault.Retry = &Retry{}
+	}
+	switch result.Vault.Retry.NumRetries {
+	case 0:
+		result.Vault.Retry.NumRetries = ctconfig.DefaultRetryAttempts
+	case -1:
+		result.Vault.Retry.NumRetries = 0
```

Comment:
Can we move this logic into `parseRetry`?

Reply:
I had it there initially, I just moved it out in the most recent commits so that we could have a consistent default behaviour even when there's no `vault{}` or no `vault.retry{}` stanza.

Reply:
That's fair. I was trying to think if we could keep the assignment/modification of `result.Vault.Retry` within its `parseRetry` func.